src/routes/schema.rs

```diff
@@ -0,0 +1,43 @@
+use rocket::{delete, get, put, State};
+use rocket::response::status;
+use rocket::serde::json::Json;
+use rocket_okapi::openapi;
+
+use crate::authn::ApiKey;
+use crate::errors::response::AgentError;
+use crate::schemas::schema::Schema;
+use crate::services::schema::SchemaStore;
+
+#[openapi]
+#[get("/schema")]
+pub async fn get_schema(
+    _auth: ApiKey,
+    schema_store: &State<Box<dyn SchemaStore>>
+) -> Result<Json<Schema>, AgentError> {
+    Ok(Json::from(schema_store.get_schema().await))
+}
+
+#[openapi]
+#[put("/schema", format = "json", data = "<schema>")]
+pub async fn update_schema(
+    _auth: ApiKey,
+    schema_store: &State<Box<dyn SchemaStore>>,
+    schema: Json<Schema>
+) -> Result<Json<Schema>, AgentError> {
+    match schema_store.update_schema(schema.into_inner()).await {
```

Comment:
Missing a check that the policy and entities stored are valid with the new schema you are trying to update, otherwise you can have a "corrupted" state of a policy and data with invalid schema